bst.py

```diff
@@ -0,0 +1,126 @@
+        return not (self.right_child or self.left_child)
+
+
+class BST(object):
+    '''Instantiate binary search tree'''
+    def __init__(self, values=None):
+        self.root = None
+        self._size = 0
+
+    def size(self):
+        '''Will return integer size of BST'''
+        return self._size
+
+    def insert(self, value):
+        '''Inserts the data in value into BST'''
+        if self.root is None:
+            self.root = BSTNode(value)
+            self._size += 1
+            return
+        current_node = self.root
+        while True:
+            if current_node.value > value:
+                if current_node.left:
+                    current_node = current_node.left
+                else:
+                    current_node.left = BSTNode(value)
+                    self._size += 1
+                    break
+            elif current_node.value < value:
+                if current_node.right:
+                    current_node = current_node.right
+                else:
+                    current_node.right = BSTNode(value)
+                    self._size += 1
+                    break
+            else:
+                break
+
+    def contains(self, value):
+        '''Returns true if data in value is in BST'''
+        if self.root is None:
+            return False
+        current_node = self.root
+        while True:
+            if current_node.value > value:
+                if current_node.left:
+                    current_node = current_node.left
+                else:
+                    return False
+            elif current_node.value < value:
+                if current_node.right:
+                    current_node = current_node.right
+                else:
+                    return False
+            else:
+                return True
+
+    def depth(self):
+        '''Returns total number of levels in BST as interger'''
+        if self.root is None:
+            return 0
+        return self._depth(1, self.root)
+
+    def _depth(self, curr_depth, local_root):
+        '''Helper function for depth'''
+        l_depth = r_depth = 0
+        if local_root.left:
+            l_depth = self._depth(curr_depth + 1, local_root.left)
+        if local_root.right:
+            r_depth = self._depth(curr_depth + 1, local_root.right)
+        return max(curr_depth, l_depth, r_depth)
+
+    def balance(self):
+        '''Return positive or negative integer to represent tree balance'''
+        ret_value = 0
+        if self.root is None:
+            return ret_value
+        if self.root.left:
+            ret_value += self._depth(1, self.root.left)
+        if self.root.right:
+            ret_value -= self._depth(1, self.root.right)
+        return ret_value
+
+
+if __name__ == '__main__':
+
+    def easy_tree():
+        x = random.sample(range(100), 100)
+        bst = BST()
+        bst.insert(50)
+        bst.insert(42.1)
+        for i in x:
+            bst.insert(i)
+        bst.contains(42.1)
```

Comment:
What does this tree look like?  What does using a random sampling of data do?  And since 42.1 is the second value inserted into the tree, is it ever anything but the first item to the left of the root node 50?